Week1/app/src/main/java/com/example/week1/db/TodoViewModel.kt

```diff
@@ -0,0 +1,64 @@
+package com.example.week1.db
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.ViewModelProvider
+import androidx.lifecycle.viewModelScope
+import kotlinx.coroutines.launch
+
+class TodoViewModel(
+    todoDao: TodoDao
+) : ViewModel() {
+
+    private val repository: TodoRepository = TodoRepository(todoDao)
+    var currentTodoList: LiveData<List<Todo>> = repository.todoList
+
+    fun insertTodo(todo: Todo) {
+        viewModelScope.launch {
+            repository.insert(todo)
+        }
+    }
+
+    fun updateTodo(todo: Todo) {
+        viewModelScope.launch {
+            repository.update(todo)
+        }
+        currentTodoList = repository.todoList
+    }
+
+    fun deleteTodo(todo: Todo) {
+        viewModelScope.launch {
+            repository.delete(todo)
+        }
+        currentTodoList = repository.todoList
+    }
+
+    fun isEntryValid(content: String): Boolean {
+        if (content.isBlank()) {
+            return false
+        }
+        return true
+    }
+
+    fun getDoneCount(): Int {
+        var count = 0
+        for (todo in requireNotNull(currentTodoList.value) { "ToDoViewModel's _currentToDoList is null" }) {
+            if(todo.isDone) count++
+        }
+
+        return count
```

Comment:
```suggestion
        return currentTodoList.filter { it.isDone }.count()
```

이런식으로 함수형 chaining으로 깔끔하게 count를 구할 수 있습니다~